app/src/main/java/org/p2p/wallet/newsend/model/SearchState.kt

```diff
@@ -24,28 +26,38 @@ data class SearchState(
                     State.ShowInvalidAddresses(foundResult)
                 foundResult.isNotEmpty() ->
                     State.UsersFound(query, foundResult)
+                isLoading && query.isNotEmpty() && foundResult.isEmpty() ->
+                    State.ShowLoadingState(query)
                 query.isNotEmpty() && foundResult.isEmpty() ->
                     State.UsersNotFound(query)
                 !recentRecipients.isNullOrEmpty() ->
-                    State.ShowRecipients(recentRecipients!!)
+                    State.ShowRecipients(recentRecipients)
                 else ->
                     State.ShowEmptyState
             }
         }
 
-    fun updateSearchResult(newQuery: String, newResult: List<SearchResult>) {
-        query = newQuery
+    fun updateLoading(isLoading: Boolean): SearchState = copy(isLoading = isLoading)
 
-        foundResult.clear()
-        foundResult += newResult
+    fun updateSearchResult(newQuery: String, newResult: List<SearchResult>): SearchState {
+        return SearchState(
+            query = newQuery,
+            foundResult = newResult,
+            recentRecipients = recentRecipients,
+            isLoading = false
+        )
     }
 
-    fun updateRecipients(newRecipients: List<SearchResult>) {
-        recentRecipients = newRecipients
+    fun updateRecipients(newRecipients: List<SearchResult>): SearchState {
+        return this.copy(recentRecipients = newRecipients, isLoading = false)
     }
 
-    fun clear() {
-        this.query = emptyString()
-        this.foundResult.clear()
+    fun reset(): SearchState {
+        return SearchState(
+            query = emptyString(),
+            foundResult = listOf(),
+            recentRecipients = recentRecipients,
+            isLoading = false
+        )
```

Comment:
No, no. 
Please don't touch the state class. 
You just need to change `showLoading` method in fragment which shows skeleton instead of progress bar.
It's not a part of the list items, it's a separate component in fragment xml.

Let's talk about the states on android meeting